clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -3512,6 +3524,227 @@ test_direct_switch()
+
+    // Zoom in and check the first sequence record by record with value checks.
+    int idx = 0;
+    bool res = true;
+    res = res &&
+        check_ref(refs[0], idx, TID_BASE, TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_VERSION) &&
+        check_ref(refs[0], idx, TID_BASE, TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_TIMESTAMP,
+                  TIMESTAMP) &&
+        check_ref(refs[0], idx, TID_BASE, TRACE_TYPE_INSTR) &&
+        check_ref(refs[0], idx, TID_BASE, TRACE_TYPE_INSTR) &&
+        check_ref(refs[0], idx, TID_BASE, TRACE_TYPE_INSTR) &&
+        // Thread switch.
+        check_ref(refs[0], idx, TID_BASE + 2, TRACE_TYPE_MARKER,
+                  TRACE_MARKER_TYPE_CONTEXT_SWITCH_START, scheduler_t::SWITCH_THREAD) &&
+        check_ref(refs[0], idx, TID_BASE + 2, TRACE_TYPE_MARKER,
+                  TRACE_MARKER_TYPE_TIMESTAMP, THREAD_SWITCH_TIMESTAMP) &&
+        check_ref(refs[0], idx, TID_BASE + 2, TRACE_TYPE_INSTR) &&
+        check_ref(refs[0], idx, TID_BASE + 2, TRACE_TYPE_INSTR) &&
+        check_ref(refs[0], idx, TID_BASE + 2, TRACE_TYPE_MARKER,
+                  TRACE_MARKER_TYPE_CONTEXT_SWITCH_END, scheduler_t::SWITCH_THREAD) &&
+        // We now see the headers for this thread.
+        check_ref(refs[0], idx, TID_BASE + 2, TRACE_TYPE_MARKER,
+                  TRACE_MARKER_TYPE_VERSION) &&
+        check_ref(refs[0], idx, TID_BASE + 2, TRACE_TYPE_MARKER,
+                  TRACE_MARKER_TYPE_TIMESTAMP, TIMESTAMP) &&
+        // The 3-instr quantum should not count the 2 switch instrs.
+        check_ref(refs[0], idx, TID_BASE + 2, TRACE_TYPE_INSTR) &&
+        check_ref(refs[0], idx, TID_BASE + 2, TRACE_TYPE_INSTR) &&
+        check_ref(refs[0], idx, TID_BASE + 2, TRACE_TYPE_INSTR) &&
+        // Process switch.
+        check_ref(refs[0], idx, TID_BASE + 4, TRACE_TYPE_MARKER,
+                  TRACE_MARKER_TYPE_CONTEXT_SWITCH_START, scheduler_t::SWITCH_PROCESS) &&
+        check_ref(refs[0], idx, TID_BASE + 4, TRACE_TYPE_MARKER,
+                  TRACE_MARKER_TYPE_TIMESTAMP, PROCESS_SWITCH_TIMESTAMP) &&
+        check_ref(refs[0], idx, TID_BASE + 4, TRACE_TYPE_INSTR) &&
+        check_ref(refs[0], idx, TID_BASE + 4, TRACE_TYPE_INSTR) &&
+        check_ref(refs[0], idx, TID_BASE + 4, TRACE_TYPE_MARKER,
+                  TRACE_MARKER_TYPE_CONTEXT_SWITCH_END, scheduler_t::SWITCH_PROCESS) &&
+        // We now see the headers for this thread.
+        check_ref(refs[0], idx, TID_BASE + 4, TRACE_TYPE_MARKER,
+                  TRACE_MARKER_TYPE_VERSION) &&
+        check_ref(refs[0], idx, TID_BASE + 4, TRACE_TYPE_MARKER,
+                  TRACE_MARKER_TYPE_TIMESTAMP, TIMESTAMP) &&
+        // The 3-instr quantum should not count the 2 switch instrs.
+        check_ref(refs[0], idx, TID_BASE + 4, TRACE_TYPE_INSTR) &&
+        check_ref(refs[0], idx, TID_BASE + 4, TRACE_TYPE_INSTR) &&
+        check_ref(refs[0], idx, TID_BASE + 4, TRACE_TYPE_INSTR) && true;
```

Comment:
Why was the `&& true` needed?

Reply:
Removed; mistakenly left when it was a placeholder during development.